src/homework_8/myArrayList/MyArrayList.java

```diff
@@ -0,0 +1,138 @@
+package homework_8.myArrayList;
+
+import java.util.Arrays;
+
+public class MyArrayList<T> implements List<T> {
+    private static final Object[] EMPTY_ARRAY = {};
+    private static final Object[] FULL_ARRAY_CAPACITY = new Object[10];
+    private T[] array;
+
+    public MyArrayList() {
+        array = (T[]) FULL_ARRAY_CAPACITY;
+    }
+
+    public MyArrayList(int capacity) {
+        if (capacity > 0) {
+            array = (T[]) new Object[capacity];
+        } else if (capacity == 0) {
+            array = (T[]) EMPTY_ARRAY;
+        } else new IllegalArgumentException("Передано невірний аргумент");
+    }
+
+    /**
+     * Окрема реалізація метода add()
+     */
+    @Override
+    public void add(T value) {
+        if (array.length != 0) {
+            boolean flag = true;
+
+            for (int i = 0; i < array.length; i++) {
+                if (array[i] == null && flag) {
+                    array[i] = value;
+                    flag = false;
+                    break;
+                }
+            }
+            if (flag) {
+                int counter = 0;
+                T[] tempArray = (T[]) new Object[array.length + (array.length >> 1)];
+
+                for (int j = 0; j < array.length; j++) {
+                    tempArray[j] = array[j];
+                    counter++;
+                }
+                array = tempArray;
+                array[counter] = value;
+            }
+        } else {
+            array = (T[]) FULL_ARRAY_CAPACITY;
+            array[0] = value;
+        }
+    }
+
+    /**
+     * Окрема реалізація метода add()
+     */
+    @Override
+    public void add(T value, int index) {
+        checkItByIndex(index);
```

Comment:
try to check how will work this method, if array.size == 10 and than call this method